src/__test__/components/plots/ViolinPlot.test.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import * as rtl from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import preloadAll from 'jest-next-dynamic';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import _ from 'lodash';
+
+import {
+  initialComponentConfigStates,
+  initialPlotConfigStates,
+} from '../../../redux/reducers/componentConfig/initialState';
+import initialExperimentState from '../../../redux/reducers/experimentSettings/initialState';
+import genes from '../../../redux/reducers/genes/initialState';
+import ViolinPlot from '../../../components/plots/ViolinPlot';
+
+jest.mock('localforage');
+const mockStore = configureMockStore([thunk]);
+
+const defaultStore = {
+  cellSets: {},
+  componentConfig: initialComponentConfigStates,
+  embeddings: {},
+  experimentSettings: {
+    ...initialExperimentState,
+  },
+  genes,
+};
+
+const experimentId = 'mockExperimentId';
+const plotUuid = 'ViolinMain'; // At some point this will stop being hardcoded
+
+describe('ViolinPlot', () => {
+  let store = null;
+
+  beforeAll(async () => {
+    await preloadAll();
+  });
+  beforeEach(() => {
+    jest.clearAllMocks(); // Do not mistake with resetAllMocks()!
```

Comment:
what is the difference between `clearAllMocks` and `resetAllMocks`?

Reply:
Potentially painful! :-)
* `clearAllMocks()` clears the couters/state of your mocks
* `resetAllMocks()` removes your mocks. Wasted a long time after using what looked right but wasn't!

Reply:
I see! So `resetAllMocks` may remove the mocks, but doesn't clean the state? That does sound like an easy trap to fall in, assuming that it will clean the state too

Reply:
No, sorry. Best thing is to check the short docs https://jestjs.io/docs/mock-function-api#mockfnmockclear rather than relying on my broken explanations :-D

My understanding is that, assuming that you are doing
```
    jest.spyOn(global.Math, 'random').mockReturnValue(MOCK_RANDOM);
```
After `clearAllMocks()`, `Math.random()` is still a function that returns `MOCK_RANDOM`, but its call history is set to zero.
After `resetAllMocks()`, `Math.random()` is not a mocked function anymore, but rather the real random thing.

Reply:
thanks for clarifying!